src/Pos/Block/Logic/Internal.hs

```diff
@@ -10,8 +10,10 @@ module Pos.Block.Logic.Internal
          -- * Constraints
          MonadBlockBase
        , MonadBlockVerify
+       , MonadBlockApplyDB
        , MonadBlockApply
 
+       , applyBlocksDbUnsafe
```

Comment:
My point is that we should separate `applyBlocksUnsafe` and `normalize` and let the caller decide whether they need to call both or only one. Because actually there is only one place where `normalize` should be done, in other places it's only needed to apply.